app/models/main_page_refresher.rb

```diff
@@ -0,0 +1,28 @@
+module MainPageRefresher
+  def prepare_data(transaction)
+    organization = transaction.organization
+    transaction.bank_account.touch
```

Comment:
@dissident not sure why do you make `touch` here. You don't make any data update here but touch is required only for update.